test/cpu_fusion.cpp

```diff
@@ -3113,6 +3114,149 @@ TEST(cpu_fusion, fuse_leaky_relu)
     EXPECT_TRUE(test::all_close(cpu2_results.at(0), expected_result));
 }
 
+TEST(cpu_fusion, fuse_update_slice)
+{
+    auto make_function = [](bool fuse = true) {
+        auto input = std::make_shared<op::Parameter>(element::f32, Shape{4, 32, 16});
+        Shape lower_bounds{1, 0, 0};
+        Shape upper_bounds{2, 32, 16};
+        auto slice = std::make_shared<op::Slice>(
+            input, fuse ? lower_bounds : Shape{3, 0, 0}, fuse ? upper_bounds : Shape{4, 32, 16});
+        auto update = std::make_shared<op::Parameter>(element::f32, Shape{1, 32, 16});
```

Comment:
Trying to understand : shouldn't we add a GOE or some other node in between so that we don't fuse?
I don't understand how changing lower and upper bounds prevent fusion.

Reply:
fusion callback should check that the slice and replace_slice are indeed working on the same portion of the tensor. Otherwise, it shouldn't fuse.